### ISSUE TITLE
[Bug] Redirected to instance not selected.

### ISSUE THREAD

**Issue by JediMaster25** [libredirect/libredirect#489]
I keep getting redirected to https://de.leddit.xyz from time to time when I use reddit even though that instance doesn't appear in the list of default instances.

**Comment by EdwardLangdon** [comment]
Maybe you need to update Instances ?

**Comment by 2084x** [comment]
Same issue here. Being redirect to instances I haven't selected. Being redirected to libreddit when I have teddit selected etc. I've tried updating instances and doing all my settings again. Gonna try a reinstall now I guess.

Arch linux - Brave - libredirect 2.3.1

**Comment by JediMaster25** [comment]
I updated instances but I have kept having the error a few times during the day.

**Comment by 2084x** [comment]
Reinstalling changed nothing, but exporting settings --> manually edit json file --> import settings got everything to save and work properly.

**Comment by JacobKfromIRC** [comment]
On version 2.3.4 this seems to still be happening. It does not seem to happen on a clean install, but an old install updated to 2.3.4 is still having the problem. I'm getting sent to domains such as searx.loafland.xyz when searching and invidious.osi.kr when YouTubing. Exporting and importing to a different Abrowser/Firefox profile also seems to fix the problem.

I suspect the problem remains on the old install because however instances are removed only removes them from the "redirects" key, whereas when actually redirecting the extension is looking at the protocol-specific key (e.g. "searxngNormalRedirectsChecks"), which doesn't exist in new installs but takes precendence over the new value.

Should I open a new issue for this?

**Comment by IkelAtomig** [comment]
No. I think better export your settings. Reset and import. That would be the only for this ?
Kai, What do you think ?

**Comment by JacobKfromIRC** [comment]
Would it be reasonable to detect this issue and then automatically silently export and re-import to fix it? That way people who don't know about or how to deal with this issue don't have to spend time troubleshooting. These are hypothetical people, so maybe it's not really that important.

**Comment by IkelAtomig** [comment]
There isn't anyway to detect and silently do this. If we do, It would increase codebase by a large scale and tedious.

We better mention in FAQ about trying resetting.

**Comment by JacobKfromIRC** [comment]
Okay, I think that is reasonable.

Is there a good way to make a backup in a way that preserves the issue? I tried copying the extension folders in storage/default but they seem to have a different name for each profile, and renaming them to fit a different profile doesn't seem to work for some reason. I guess copying the entire profile root folder should be sufficient, but I'm not sure how to check that.
